src/mimefactory.rs

```diff
@@ -500,18 +501,21 @@ impl<'a> MimeFactory<'a> {
                 }
 
                 if location::is_sending_locations_to_chat(context, self.msg.chat_id) {
-                    if let Ok((kml_file, last_added_location_id)) =
-                        location::get_kml(context, self.msg.chat_id)
-                    {
-                        wrapmime::add_filename_part(
-                            message,
-                            "location.kml",
-                            "application/vnd.google-earth.kml+xml",
-                            &kml_file,
-                        )?;
-                        if !self.msg.param.exists(Param::SetLatitude) {
-                            // otherwise, the independent location is already filed
-                            self.out_last_added_location_id = last_added_location_id;
+                    match location::get_kml(context, self.msg.chat_id) {
+                        Ok((kml_content, last_added_location_id)) => {
+                            wrapmime::add_filename_part(
+                                message,
+                                "location.kml",
+                                "application/vnd.google-earth.kml+xml",
+                                &kml_content,
+                            )?;
+                            if !self.msg.param.exists(Param::SetLatitude) {
+                                // otherwise, the independent location is already filed
+                                self.out_last_added_location_id = last_added_location_id;
+                            }
+                        }
+                        Err(err) => {
+                            warn!(context, "mimefactory: could not get location: {}", err);
```

Comment:
Do we really want to display this to user? I don't know how interaction with user application (eg. android) works here. What happens if location is not available from the OS, but user enabled location streaming? Will the app start displaying these warnings?

Reply:
only errors are displayed, warning go to the log file.